pkg/resolver/related.go

```diff
@@ -22,3 +32,301 @@ type SearchRelatedResult struct {
 // 	klog.Info("TODO: Resolve SearchRelatedResult: Items() - model/related.go")
 // 	return nil
 // }
+func (s *SearchResult) buildRelationsQuery() {
+	// Example query to find relations between resources - accepts an array of uids
+	// =============================================================================
+	// 	SELECT "uid", "kind", MIN("level") AS "level" FROM
+	// (
+	// 	SELECT "level", unnest(array[sourceid, destid, concat('cluster__',cluster)]) AS "uid",
+	// unnest(array[sourcekind, destkind, 'Cluster']) AS "kind"
+	// 	FROM (
+	// 		WITH RECURSIVE search_graph(level, sourceid, destid,  sourcekind, destkind, cluster) AS
+	// 		(SELECT 1 AS "level", "sourceid", "destid", "sourcekind", "destkind", "cluster"
+	// 		 FROM "search"."all_edges" AS "e"
+	// 		 WHERE (("destid" IN ('local-cluster/108a77a2-159c-4621-ae1e-7a3649000ebc' )) OR
+	// 					("sourceid" IN ('local-cluster/108a77a2-159c-4621-ae1e-7a3649000ebc'))
+	// 			   )
+	// 				   UNION
+	// 		 (SELECT level+1 AS "level", "e"."sourceid", "e"."destid", "e"."sourcekind", "e"."destkind", "e"."cluster"
+	// 		  FROM "search"."all_edges" AS "e"
+	// 		  INNER JOIN "search_graph" AS "sg"
+	// 		  ON (("sg"."destid" IN ("e"."sourceid", "e"."destid")) OR
+	// 			  ("sg"."sourceid" IN ("e"."sourceid", "e"."destid"))
+	// 			 )
+	//  		  WHERE (("e"."destkind" != 'Node') AND
+	// 				 ("sg"."level" <=3)
+	//  				)
+	// 		 )
+	// 		) SELECT DISTINCT "level", "sourceid", "destid", "sourcekind", "destkind", "cluster" FROM "search_graph"
+	// 	) AS "search_graph"
+	// ) AS "combineIds"
+	// WHERE (("level" <=3)
+	// AND ("uid" NOT IN ('local-cluster/108a77a2-159c-4621-ae1e-7a3649000ebc')))
+	// GROUP BY "uid", "kind"
+	// -- union -- This is added if `kind:Cluster` is present in search term
+	// -- select uid as uid, data->>'kind' as kind, 1 AS "level" FROM search.resources where cluster IN ('local-cluster')
+
+	var whereDs []exp.Expression
+	// This level will come into effect only in case of Application/Subscription relations.
+	// For normal searches, we go only upto level 1. This can be changed later, if necessary.
+	level := config.Cfg.RelationLevel
+	//The level can be parameterized later, if needed
+	whereDs = append(whereDs, goqu.C("level").Lte(level))  // Add filter to select upto level (default 3) relationships
+	whereDs = append(whereDs, goqu.C("uid").NotIn(s.uids)) // Add filter to avoid selecting the search object itself
```

Comment:
You should be able to simplify this one too.
```suggestion
	// This level will come into effect only in case of Application/Subscription relations.
	// For normal searches, we go only up to level 1. This can be changed later, if necessary.
	level := config.Cfg.RelationLevel
	
	// The level can be parameterized later, if needed
	whereDs := []exp.Expression{
		goqu.C("level").Lte(level))     // Add filter to select up to level (default 3) relationships
		goqu.C("uid").NotIn(s.uids)) // Add filter to avoid selecting the search object itself
```

Reply:
Done